src/cairoUtilFuncGen/memory/staticIndexAccess.ts

```diff
@@ -30,20 +34,28 @@ export class MemoryStaticArrayIndexAccessGen extends CairoUtilFuncGenBase {
       ...WM_INDEX_STATIC,
       [
         ['arr', typeNameFromTypeNode(arrayType, this.ast), DataLocation.Memory],
-        ['index', createUint256TypeName(this.ast)],
-        ['width', createUint256TypeName(this.ast)],
-        ['length', createUint256TypeName(this.ast)],
+        ['index', createFeltTypeName(this.ast)],
+        ['width', createFeltTypeName(this.ast)],
+        ['length', createFeltTypeName(this.ast)],
       ],
       [['child', typeNameFromTypeNode(arrayType.elementT, this.ast), DataLocation.Memory]],
     );
+
+    const u256ToFelt252 = this.ast.registerImport(
+      indexAccess,
+      ...U256_TO_FELT252,
+      [['x', createUint256TypeName(this.ast)]],
+      [['r', createFeltTypeName(this.ast)]],
+    );
+
     const width = CairoType.fromSol(arrayType.elementT, this.ast).width;
     return createCallToFunction(
       importFunc,
       [
         indexAccess.vBaseExpression,
-        indexAccess.vIndexExpression,
-        createNumberLiteral(width, this.ast, 'uint256'),
-        createNumberLiteral(arrayType.size, this.ast, 'uint256'),
+        createCallToFunction(u256ToFelt252, [indexAccess.vIndexExpression], this.ast),
+        createNumberLiteral(width, this.ast, 'uint252'),
```

Comment:
A constant with a comment what `uint252` exactly is and why it was introduced would be useful. I've seen you've added functions like createtypename and it's good. However since it's unintuitive and there is also `felt252` it should be explained why it is needed and also relation between the two